plugin-server/src/utils/db/db.ts

```diff
@@ -678,19 +720,90 @@ export class DB {
             res[`group${index}_properties`] = JSON.stringify(properties)
         }
 
-        for (const gi of notHandledIdentifiers) {
-            void this.updateGroupPropertiesCache(teamId, gi.index, gi.key, {})
-            res[`group${gi.index}_properties`] = JSON.stringify({}) // also adding to event for consistency
+        for (const groupIdentifier of notHandledIdentifiers) {
+            void this.updateGroupPropertiesCache(teamId, groupIdentifier.index, groupIdentifier.key, {})
+            res[`group${groupIdentifier.index}_properties`] = JSON.stringify({}) // also adding to event for consistency
+        }
+        return res
+    }
+
+    public async getGroupsCreatedAtFromDbAndUpdateCache(
```

Comment:
This function is doing two things rather than 1:
- SQL building
- Querying and parsing results

Suggestion: pull the SQL building out to a separate function and create a test for it?